sklearn/neural_network/extreme_learning_machines.py

```diff
@@ -0,0 +1,629 @@
+                 random_state):
+        self.C = C
+        self.activation = activation
+        self.class_weight = class_weight
+        self.weight_scale = weight_scale
+        self.batch_size = batch_size
+        self.n_hidden = n_hidden
+        self.verbose = verbose
+        self.warm_start = warm_start
+        self.random_state = random_state
+
+    def _init_weights(self, n_features):
+        """Initialize the parameter weights."""
+        rng = check_random_state(self.random_state)
+
+        # Use the initialization method recommended by Glorot et al.
+        weight_init_bound = np.sqrt(6. / (n_features + self.n_hidden))
+
+        self.coef_hidden_ = rng.uniform(-weight_init_bound,
+                                        weight_init_bound, (n_features,
+                                                            self.n_hidden))
+        self.intercept_hidden_ = rng.uniform(-weight_init_bound,
+                                             weight_init_bound,
+                                             self.n_hidden)
+        if self.weight_scale != 1:
+            self.coef_hidden_ *= self.weight_scale
+            self.intercept_hidden_ *= self.weight_scale
+
+    def _compute_hidden_activations(self, X):
+        """Compute the hidden activations."""
+
+        hidden_activations = safe_sparse_dot(X, self.coef_hidden_)
+        hidden_activations += self.intercept_hidden_
+
+        # Apply the activation method
+        activation = ACTIVATIONS[self.activation]
+        hidden_activations = activation(hidden_activations)
+
+        return hidden_activations
+
+    def _fit(self, X, y, sample_weight=None, incremental=False):
+        """Fit the model to the data X and target y."""
+        # Validate input params
+        if self.n_hidden <= 0:
+            raise ValueError("n_hidden must be > 0, got %s." % self.n_hidden)
+        if self.C <= 0.0:
+            raise ValueError("C must be > 0, got %s." % self.C)
+        if self.activation not in ACTIVATIONS:
+            raise ValueError("The activation %s is not supported. Supported "
+                             "activation are %s." % (self.activation,
+                                                     ACTIVATIONS))
+
+        # Initialize public attributes
+        if not hasattr(self, 'classes_'):
+            self.classes_ = None
+        if not hasattr(self, 'coef_hidden_'):
+            self.coef_hidden_ = None
+
+        # Initialize private attributes
+        if not hasattr(self, '_HT_H_accumulated'):
+            self._HT_H_accumulated = None
+
+        X, y = check_X_y(X, y, accept_sparse=['csr', 'csc', 'coo'],
+                         dtype=np.float64, order="C", multi_output=True)
+
+        # This outputs a warning when a 1d array is expected
+        if y.ndim == 2 and y.shape[1] == 1:
+            y = column_or_1d(y, warn=True)
+
+        # Classification
+        if isinstance(self, ClassifierMixin):
```

Comment:
This looks ugly. There are separate classes, right? Why not add a member ``encode_y`` that is identity for regression and does this in classification? This shouldn't live in the general base-class. I think there is a function like that in GBRT, maybe we should give the same name here.